crates/guest-rust/src/cabi_realloc.c

```diff
@@ -0,0 +1,8 @@
+#include <stdint.h>
+
+extern void *cabi_realloc_wit_bindgen_0_18_0(void *ptr, size_t old_size, size_t align, size_t new_size);
+
+__attribute__((__weak__, __export_name__("cabi_realloc")))
+void *cabi_realloc(void *ptr, size_t old_size, size_t align, size_t new_size) {
+  return cabi_realloc_wit_bindgen_0_18_0(ptr, old_size, align, new_size);
+}
```

Comment:
If we're doing the `_0_18_0`  mangling anyway, would it be simpler to teach all the places that name `cabi_realloc` to name `cabi_realloc_0_18_0` instead, rather than using a precompiled archive to get a weak symbol?

Reply:
Right now that's a hardcoded convention in `wasm-tools component new`, there's nothing in the actual core wasm binary produce that refers to `cabi_realloc` at all currently.